components/ReferenceManager/form/ReferenceItemRhUserLookupInput.tsx

```diff
@@ -0,0 +1,127 @@
+import { isEmpty } from "~/config/utils/nullchecks";
+import { fetchUserSuggestions } from "~/components/SearchSuggestion/lib/api";
+import { ID, NullableString } from "~/config/types/root_types";
+import { ReactElement, ReactNode, SyntheticEvent, useState } from "react";
+import { SuggestedUser } from "~/components/SearchSuggestion/lib/types";
+import Autocomplete from "@mui/material/Autocomplete";
+import ReferenceItemRhUserLookupInputTag from "./ReferenceItemRhUserLookupInputTag";
+import TextField from "@mui/material/TextField";
+
+export type InputProps = {
+  disabled?: boolean;
+  filterUserIDs: ID[];
+  label?: NullableString;
+  onUserSelect?: (user: SuggestedUser) => void;
+  placeholder?: string;
+  required?: boolean;
+  selectedUser?: SuggestedUser;
+  shouldClearOnSelect?: boolean;
+};
+
+const getDebouncedHandleInputChange = ({
+  debounceTime = 500,
+  onLoad,
```

Comment:
Usually for calls like this, a given component needs to perform a side-effect while the call / debounce function is happening in the background. In this case, onLoad is a callback to recognize this before completing a given function of  the method. I renamed it to `onFetchLoad` to avoid confusion